### PR TITLE
New Recipe: BVLS (fortran)

### PR DESCRIPTION
**Pull request by kose-y**

### CODE REVIEW

File: B/BVLS/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,28 @@
+using BinaryBuilder, Pkg
+
+name = "BVLS"
+version = v"1990.03.19"
```
Comment (giordano) [inline B/BVLS/build_tarballs.jl:4]: Where does this version number come from?  Is that a date?  The file mentions the 2007 year

**Comment by giordano** [comment]
Also, where does the source come from?  It seems to have an open license, so looks good, I just find it curious there is no better option than vendoring the file itself

File: B/BVLS/build_tarballs.jl (hunk 1 above)
Comment (kose-y) [reply B/BVLS/build_tarballs.jl:4]: [Line 10 of bvls.f](https://github.com/JuliaPackaging/Yggdrasil/pull/4210/files#diff-47ec748f26cb959b0367ab8492eafeba4d1e4e9d05c26dabe98aee50d42630a7R10). Seems to be a date.

Comment (giordano) [reply B/BVLS/build_tarballs.jl:4]: [Line 18](https://github.com/JuliaPackaging/Yggdrasil/pull/4210/files#diff-47ec748f26cb959b0367ab8492eafeba4d1e4e9d05c26dabe98aee50d42630a7R18) says November 2007 though, as I said above

**Comment by giordano** [comment]
I found http://lib.stat.cmu.edu/general/bvls which looks older (no reference to 2007), but crucially it had a non open source license.  I'm more and more interested in knowing where this file is actually coming from

**Comment by kose-y** [comment]
This code is bundled inside a [CRAN package bvls](https://cran.r-project.org/web/packages/bvls/index.html). 2007 is when the first version of this package came out, and the inst/COPYRIGHTS file inside the package states:
```
src/bvls.f
  From the source code accompanying the article 

  Stark PB, Parker RL (1995). Bounded-variable least-squares:
  an algorithm and applications, Computational Statistics, 10, 129-141,

  and downloaded from http://lib.stat.cmu.edu/general/bvls The code
  was modified slightly to allow compatibility with gfortran.  The
  authors have agreed to allow distribution under GPL version 2 or
  newer.
```

**Comment by giordano** [comment]
Perfect, thanks for the explanation!  What about using https://cran.r-project.org/src/contrib/bvls_1.4.tar.gz as [`ArchiveSource`](https://docs.binarybuilder.org/stable/reference/#BinaryBuilderBase.ArchiveSource)?  I'd prefer that to vendoring the source code here, if possible :slightly_smiling_face:

**Comment by kose-y** [comment]
Sure, I will try.

**Comment by kose-y** [comment]
Now updated to use ArchiveSource.

**Comment by giordano** [comment]
Lovely, thanks!

**Comment by kose-y** [comment]
Thank you very much!

**Pull request merged**